src/test/java/seedu/internbuddy/logic/commands/UpdateCommandTest.java

```diff
@@ -0,0 +1,99 @@
+        updatedApplications.add(newApplication);
+
+        Company updatedCompany = new Company(companyToUpdate.getName(), companyToUpdate.getPhone(),
+                companyToUpdate.getEmail(), companyToUpdate.getAddress(), companyToUpdate.getTags(),
+                companyToUpdate.getStatus(), updatedApplications, companyToUpdate.getIsFavourite());
+
+        model.setCompany(companyToUpdate, updatedCompany);
+
+        AppStatus newAppStatus = new AppStatus("INTERVIEWED");
+
+        UpdateCommand updateCommand = new UpdateCommand(INDEX_FIRST_COMPANY, INDEX_FIRST_COMPANY, newAppStatus);
+
+        Application applicationToUpdate = updatedCompany.getApplications().get(0);
+        Application updatedApplication = applicationToUpdate.setAppStatus(newAppStatus);
+
+        CommandResult result = updateCommand.execute(model);
+
+        assertEquals(String.format(UpdateCommand.MESSAGE_UPDATE_APPLICATION_SUCCESS, updatedApplication),
+                result.getFeedbackToUser());
+    }
+
+    @Test
+    public void execute_invalidCompanyIndex_throwsCommandException() {
+        AppStatus newAppStatus = new AppStatus("INTERVIEWED");
+        UpdateCommand updateCommand = new UpdateCommand(Index.fromOneBased(999),
+                Index.fromOneBased(1), newAppStatus);
+        assertThrows(CommandException.class, () -> updateCommand.execute(model),
+                Messages.MESSAGE_INVALID_COMPANY_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_invalidApplicationIndex_throwsCommandException() {
+        AppStatus newAppStatus = new AppStatus("INTERVIEWED");
+        UpdateCommand updateCommand = new UpdateCommand(INDEX_FIRST_COMPANY,
+                Index.fromOneBased(999), newAppStatus);
+        assertThrows(CommandException.class, () -> updateCommand.execute(model),
+                Messages.MESSAGE_INVALID_APPLICATION_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void equals_sameObject_returnsTrue() {
+        AppStatus appStatus = new AppStatus("INTERVIEWED");
+        UpdateCommand updateCommand = new UpdateCommand(INDEX_FIRST_COMPANY,
+                Index.fromOneBased(1), appStatus);
+        UpdateCommand updateCommand2 = new UpdateCommand(INDEX_FIRST_COMPANY,
+                Index.fromOneBased(1), appStatus);
+
+        // same object -> returns true
+        assertEquals(updateCommand, updateCommand2);
+    }
+
+    @Test
+    public void equals_differentObject_returnsFalse() {
+        AppStatus appStatus1 = new AppStatus("INTERVIEWED");
+        AppStatus appStatus2 = new AppStatus("OFFERED");
+        UpdateCommand updateCommand1 = new UpdateCommand(INDEX_FIRST_COMPANY,
+                Index.fromOneBased(1), appStatus1);
+        UpdateCommand updateCommand2 = new UpdateCommand(INDEX_SECOND_COMPANY,
+                Index.fromOneBased(1), appStatus2);
+
+        // different objects -> returns false
+        assertNotEquals(updateCommand1, updateCommand2);
+    }
```

Comment:
Good test cases, but I think the `equals_sameObject_returnsTrue` is not exactly testing equality for the same object but rather the another `UpdateCommandTest` instance with the same values; we should test for both. 

Would including a testing for different individual parameters be more concrete in `equals_differentObject_returnsFalse`? 

Perhaps unifying the `equals` test into one test with the 3 different cases could be better?